### PR TITLE
Update the webauthn IDL file + test

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

File: webauthn/idlharness.https.any.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+promise_test(() => {
+  const execute_test = () => idl_test(
+    ['webauthn'],
+    ['credential-management'],
+    idlArray => {
+      idlArray.add_objects({
+        WebAuthentication: ['navigator.authentication'],
+        PublicKeyCredential: ['cred'],
+        AuthenticatorAssertionResponse: ['assertion']
```
Comment (lukebjerring) [inline webauthn/idlharness.https.any.js]: @jcjones @apowers313 - I'm not sure how to check whether I've set up these tests correctly.

In short, these two fields `self.cred` and `self.assertion` are supposed to be a `PublicKeyCredential` and a `AuthenticatorAssertionResponse` respectively. Is that what you would expect to result from the code below? I can see the prompt in FF61, but how do I add a security token to Firefox?

**Comment by apowers313** [comment]
cc: @christiaanbrand @kpaulh

**Comment by kpaulh** [comment]
@lukebjerring it definitely is on the radar. We have a virtual authenticator for our own layout tests (https://cs.chromium.org/chromium/src/third_party/WebKit/LayoutTests/http/tests/credentialmanager/resources/virtual-navigator-credentials.js), but Firefox, MS, and Chrome would have to agree on a uniform testing API to do the same for WPTs. (Similar to WebBluetooth's testing API)

**Review by apowers313 — APPROVED** [review]
I reviewed against the current spec and it looks good to me. Need to make sure that browsers are planning on implementing the `tokenBinding` "not-supported" change since this is based against the latest editorial spec and not the CR version and it may impact server implementations.

cc: @akshayku @kpaulh @jcjones

**Comment by apowers313** [comment]
@lukebjerring we had talked about creating a mock authenticator framework, but it hasn't come to fruition yet. Right now, I think each vendor has it's own soft authentication (e.g. - FF has a flag for enabling it's internal soft authenticator).

**Comment by lukebjerring** [comment]
Took a stab at an automated version of the same test:
https://chromium-review.googlesource.com/c/chromium/src/+/1144133

@kpaulh can you take a look at that and we'll continue the discussion there?

**Pull request merged**